addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -156,29 +234,47 @@ private boolean isSilentRollerShutter(SomfyTahomaDevice device) {
         return false;
     }
 
+    private boolean isOnOffHeatingSystem(SomfyTahomaDevice device) {
+        SomfyTahomaDeviceDefinition def = device.getDefinition();
+        for (SomfyTahomaDeviceDefinitionCommand cmd : def.getCommands()) {
+            if (cmd.getCommandName().equals(COMMAND_SET_HEATINGLEVEL)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     private void deviceDiscovered(SomfyTahomaDevice device, ThingTypeUID thingTypeUID) {
         deviceDiscovered(device.getLabel(), device.getDeviceURL(), device.getOid(), thingTypeUID);
     }
 
     private void deviceDiscovered(String label, String deviceURL, String oid, ThingTypeUID thingTypeUID) {
-        Map<String, Object> properties = new HashMap<>(1);
+        Map<String, Object> properties = new HashMap<>(2);
```

Comment:
what is the purpose of artificially restricting the size of the hashmap?

Reply:
Changed to generic